src/components/google-maps-aerial.js

```diff
@@ -0,0 +1,144 @@
+    maxDistance: { type: 'number', default: 20000 },
+    ellipsoidalHeight: { type: 'number', default: 0 },
+    copyrightEl: { type: 'selector' }
+  },
+
+  init: function () {
+    this.tick = AFRAME.utils.throttleTick(this.tick, 10, this);
+    this.initialized = false;
+    // Initialize tiles
+    this.tiles = new TilesRenderer(
+      'https://tile.googleapis.com/v1/3dtiles/root.json'
+    );
+
+    // Register plugins
+    this.tiles.registerPlugin(
+      new GoogleCloudAuthPlugin({ apiToken: this.data.apiToken })
+    );
+    this.tiles.registerPlugin(new TileCompressionPlugin());
+    this.tiles.registerPlugin(new TilesFadePlugin());
+    this.tiles.registerPlugin(
+      new GLTFExtensionsPlugin({
+        dracoLoader: new DRACOLoader().setDecoderPath(
+          'https://unpkg.com/three@0.153.0/examples/jsm/libs/draco/gltf/'
+        )
+      })
+    );
+
+    // Set location
+    this.tiles.setLatLonToYUp(
+      this.data.latitude * MathUtils.DEG2RAD,
+      this.data.longitude * MathUtils.DEG2RAD
+    );
+
+    // Create a child entity for the height offset
+    const offsetEl = document.createElement('a-entity');
+    offsetEl.object3D.position.y = -this.data.ellipsoidalHeight;
+    offsetEl.object3D.add(this.tiles.group);
+    this.el.appendChild(offsetEl);
+    this.offsetEl = offsetEl;
+
+    // Get camera and renderer
+    this.camera =
+      this.data.cameraEl?.object3D.children[0] ??
+      document.querySelector('a-scene').camera;
+    this.renderer = this.el.sceneEl.renderer;
+    // if (!this.camera) {
+    //   throw new Error('3D Tiles: Please add an active camera or specify the target camera via the cameraEl property');
+    // }
+    this.originalCamera = this.camera;
+
+    this.tiles.setResolutionFromRenderer(this.camera, this.renderer);
+    this.tiles.setCamera(this.camera);
+    this.tiles.update();
+    this.initialized = true;
+
+    // Add this to your component's init:
+    this.el.addEventListener('cameraChange', (e) => {
+      if (e.detail && this.initialized) {
+        const prevCamera = this.camera;
+        this.camera = e.detail;
+
+        // Delete previous camera from tiles renderer first
+        if (prevCamera) {
+          this.tiles.deleteCamera(prevCamera);
+        }
+        // Set new camera and update resolution
+        this.tiles.setCamera(this.camera);
+        this.tiles.setResolutionFromRenderer(this.camera, this.renderer);
+        this.tiles.update();
+      }
+    });
+
+    if (AFRAME.INSPECTOR && AFRAME.INSPECTOR.opened) {
+      // set active inspector camera
+      this.camera = AFRAME.INSPECTOR.camera;
+      // emit play event to start load tiles in aframe-inspector
+      this.play();
+    }
+  },
+
+  tick: function () {
+    if (this.initialized && this.tiles && this.camera) {
+      // Ensure camera is set on each tick
+      this.tiles.setCamera(this.camera);
+      this.tiles.setResolutionFromRenderer(this.camera, this.renderer);
+      this.tiles.update();
+    }
```

Comment:
If you do that in tick, then you can remove all the code related to `cameraChange` event and just use 
`this.el.sceneEl.camera` instead of `this.camera` I think.

Reply:
I tried only calling tiles.update() when load-model event is fired, but load-model only seems to fire when tiles.update() is called every tick so it doesn't work as expected. It's ok to leave like this for now imho